api/team.go

```diff
@@ -409,14 +409,12 @@ func findTeams(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	} else {
 		teams := result.Data.([]*model.Team)
-
-		s := make([]string, 0, len(teams))
-
+		m := make(map[string]*model.Team)
 		for _, v := range teams {
-			s = append(s, v.Name)
+			m[v.Id] = v
 		}
 
-		w.Write([]byte(model.ArrayToJson(s)))
+		w.Write([]byte(model.TeamMapToJson(m)))
```

Comment:
I'm not sure returning the full team objects is a good idea since it includes a whole bunch more information than is needed. For example `AllowedDomains` will provide information on what domains can be used to sign up, `Type` will let on to whether this is an open or invite team, and `Email` will give away the email of the team creator.

So anyone who knows the email of a user in a team can freely get that information about the team. It would be a little different if the `findTeams` API required authentication, but as is this seems like a security issue to me.